containers/allan_dip_bot/run_bot.py

```diff
@@ -274,7 +281,18 @@ def main() -> None:
     parser.add_argument(
         "--outdir", type=Path, help="output directory for game json to be stored"
     )
+    parser.add_argument(
+        "--outdir", type=str, help="output directory for game json to be stored"
+    )
+    parser.add_argument(
+        "--aggressiveness",
+        type=str,
+        choices=["A", "M", "F"],
+        default="M",
```

Comment:
You can handle the `str`-to-`enum` conversion directly:

```suggestion
        type=Aggressiveness,
        choices=Aggressiveness.__members__,
        default=Aggressiveness.moderate.value,
```

This is just based on some other code I read recently, so it might nee some tweaking.

Reply:
I tried this to implement this for about an hour, but unfortunately __members__ doesn't work - even when entering the exact text it doesn't recognize the enum. Further, argparse warns against passing in the enum type for the options [field](https://docs.python.org/3/library/argparse.html#action:~:text=Use%20of%20enum.Enum%20is%20not%20recommended%20because%20it%20is%20difficult%20to%20control%20its%20appearance%20in%20usage%2C%20help%2C%20and%20error%20messages):

Reply:
You should uninstall the changes to the requirements you made. `enum` and `typing` are already in the standard library. The installed packages might be shadowing the proper versions.

Reply:
done

Reply:
I've tried a few more permutations. The suggestion in https://bugs.python.org/issue25061 is interesting and reduces the lines of code, but is still fundamentally taking in a string from argparse

Reply:
Weird. I might look into it later myself out of curiosity, but it's probably best to not look further into this for now.